html/webappapis/system-state-and-capabilities/the-navigator-object/protocol.https.html

```diff
@@ -170,45 +109,75 @@
   'view-source',
   'ws',
   'wss',
-  'wyciwyg');
-
-for ( var bi=0, bl=blacklist.length; bi<bl; ++bi ){
-
-  test(function () {
-    assert_throws('SECURITY_ERR', function () { navigator.registerProtocolHandler(blacklist[bi], location.href + '/%s', 'foo') } );
-  }, 'attempting to override the ' + blacklist[bi] + ' protocol should throw SECURITY_ERR');
-
+  'wyciwyg',
+  /*other invalid schemes*/
+  'unrecognized',
+  'mаilto', /*a cyrillic "а"*/
+  'mailto:',
+  'mailto://',
+  'mailto' + String.fromCharCode(0),
+  'mailtoo' + String.fromCharCode(8),
+  'mailto' + String.fromCharCode(10),
+  'http://',
+  'ssh:/',
+  'magnet:+',
+  'tel:sip',
+  'web+',
+];
+for (const scheme of blacklist) {
+  test(() => {
+    assert_throws('SECURITY_ERR', () => { navigator.registerProtocolHandler(scheme, location.href + '/%s', 'foo'); });
+  }, 'registerProtocolHandler: Attempting to override the "' + scheme + '" protocol should throw SECURITY_ERR.');
+
+  test(() => {
+    assert_throws('SECURITY_ERR', () => { navigator.unregisterProtocolHandler(scheme, location.href + '/%s', 'foo'); });
+  }, 'unregisterProtocolHandler: Attempting to override the "' + scheme + '" protocol should throw SECURITY_ERR.');
 }
 
 /* The following protocols must be possible to override.
  * We're just testing that the call goes through here. Whether or not they
  * actually work as handlers is covered by the interactive tests. */
-var whitelist = new Array(
+
+const whitelist = [
```

Comment:
Can we rename the variable to safelist while doing this refactoring as well?

(Looks great overall btw, thanks for simplifying this!)

Reply:
Sure, done.